File: N/2367-number-of-arithmetic-triplets/main.go
package main

func arithmeticTriplets(nums []int, diff int) int {
	ret := 0
	m := map[int]int{}
	for _, n := range nums {
		m[n] = 1
	}
	for i := 0; i < len(nums)-2; i++ {
		n2, n3 := nums[i]+diff, nums[i]+diff+diff
		_, ok2 := m[n2]
		_, ok3 := m[n3]
		if ok2 && ok3 {
			ret++
		}
	}
	return ret
}

File: N/2367-number-of-arithmetic-triplets/main.rs
struct Solution {}

use std::collections::HashSet;
impl Solution {
  pub fn arithmetic_triplets(nums: Vec<i32>, diff: i32) -> i32 {
    let set: HashSet<i32> = nums.clone().into_iter().collect::<HashSet<i32>>();
    nums
      .iter()
      .map(
        |&v| match set.contains(&(v + diff)) && set.contains(&(v + diff + diff)) {
          true => 1,
          _ => 0,
        },
      )
      .collect::<Vec<i32>>()
      .iter()
      .sum()
  }
}

fn main() {
  println!("{}", Solution::arithmetic_triplets(vec![1, 2, 3], 1));
}
